### PR TITLE
Add a node analyzer for code actions

### LINKED ISSUE #29474: "Document this" code action not available for module-var-decl
In language, we support documentation for `module-var-decl`. I think it is nice to have the document this code action for `module-var-decl` if someone needs to document it.

e.g.
```bal
http:Client mockyClient = check new ("https://run.mocky.io/v3"); // Document this code action not available

final int threshold = 10; // Document this action not available
```

### LINKED ISSUE #33072: `Document This` codeaction for class is suggested within object method body
**Description:**
Consider the source:
```ballerina
class Student {

    public function getName() returns string {
        
        return "name";
    }
}
```
Put the cursor on the `return` statement and you will see the `Document This` codeaction, Click on it, and the documentation will be added to the class `Student`.

**Steps to reproduce:**
See description

**Affected Versions:**
SL Beta 3 RC5

### PR DESCRIPTION
**Pull request by nadeeshaan**
Fixes #33072 
Fixes #29474

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [x] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionNodeAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+import org.ballerinalang.langserver.codeaction.NodeBasedPositionDetailsImpl.PositionDetailsBuilder;
+
+import java.util.Optional;
+
+/**
+ * Node analyzer for the code actions.
+ * This will calculate the node information and fill the {@link PositionDetailsBuilder} and the information will be
+ * reflected to the code actions via the PositionDetails API.
+ *
+ * @since 2201.0.x
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionNodeAnalyzer.java:43]: For my knowledge. Will `2201.0.x` be fine here?

**Review by IMS94 — DISMISSED** [review]
LGTM!

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionNodeAnalyzer.java (hunk 1 above)
Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionNodeAnalyzer.java:43]: Should be the update version right? I'll change this

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionNodeAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+        if (node.functionBody().kind() != SyntaxKind.FUNCTION_BODY_BLOCK) {
+            return;
+        }
+
+        NodeList<Token> qualifiers = node.qualifierList();
+        int startOffset = !qualifiers.isEmpty() ?
+                qualifiers.get(0).textRange().startOffset() :
+                node.functionKeyword().textRange().startOffset();
+        int openBraceEnd = ((FunctionBodyBlockNode) node.functionBody()).openBraceToken().textRange().endOffset();
+        int closeBraceEnd = ((FunctionBodyBlockNode) node.functionBody()).closeBraceToken().textRange().endOffset();
```
Comment (gayalkuruppu) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionNodeAnalyzer.java:82]: We can get `(FunctionBodyBlockNode) node.functionBody()` to a variable and use it so this calculation is not repeated twice

Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionNodeAnalyzer.java]: Done

**Review by mohanvive — APPROVED** [review]

**Pull request merged**